qldbdriver/qldbdriver.go

```diff
@@ -69,24 +69,23 @@ func New(ledgerName string, qldbSession *qldbsession.QLDBSession, fns ...func(*D
 	semaphore := sync2.NewSemaphore(int(options.MaxConcurrentTransactions), 0)
 	sessionPool := make(chan *session, options.MaxConcurrentTransactions)
 	isClosed := false
+	retryPolicy := RetryPolicy{MaxRetryLimit: 4, Backoff: ExponentialBackoffStrategy{SleepBase: 10000000, SleepCap: 5000000000}}
 
 	return &QLDBDriver{ledgerName, qldbSession, options.RetryLimit, options.MaxConcurrentTransactions, logger, isClosed,
-		semaphore, sessionPool}
+		semaphore, sessionPool, retryPolicy}
 }
 
-// Execute a provided function within the context of a new QLDB transaction.
-//
-// The provided function might be executed more than once.
-// It is recommended for it to be idempotent, so that it doesn't have unintended side effects in the case of retries.
-func (driver *QLDBDriver) Execute(ctx context.Context, fn func(txn Transaction) (interface{}, error)) (interface{}, error) {
-	return driver.ExecuteWithRetryPolicy(ctx, fn, RetryPolicy{MaxRetryLimit: 4, Backoff: ExponentialBackoffStrategy{SleepBaseInMillis: 10, SleepCapInMillis: 5000}})
+// SetRetryPolicy sets the driver's retry policy for Execute
+func (driver *QLDBDriver) SetRetryPolicy(rp RetryPolicy) {
+	driver.retryPolicy = rp
 }
 
-// Execute a provided function within the context of a new QLDB transaction with a custom RetryPolicy.
+// Execute a provided function within the context of a new QLDB transaction.
 //
 // The provided function might be executed more than once.
+// This function is not expected to ever run concurrently.
```

Comment:
I would say that the provided function is not expected to run concurrently. 

Your current wording implies that Execute isn't supporting concurrency which is wrong.